client/keys/add.go

```diff
@@ -165,77 +170,75 @@ type NewKeyBody struct {
 }
 
 // add new key REST handler
-func AddNewKeyRequestHandler(w http.ResponseWriter, r *http.Request) {
-	var kb keys.Keybase
-	var m NewKeyBody
+func AddNewKeyRequestHandler(indent bool) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		var kb keys.Keybase
+		var m NewKeyBody
 
-	kb, err := GetKeyBase()
-	if err != nil {
-		w.WriteHeader(500)
-		w.Write([]byte(err.Error()))
-		return
-	}
+		kb, err := GetKeyBase()
+		if err != nil {
+			w.WriteHeader(500)
+			w.Write([]byte(err.Error()))
+			return
+		}
 
-	body, err := ioutil.ReadAll(r.Body)
-	err = json.Unmarshal(body, &m)
+		body, err := ioutil.ReadAll(r.Body)
+		if err != nil {
+			w.WriteHeader(http.StatusBadRequest)
+			w.Write([]byte(err.Error()))
+			return
+		}
+		err = json.Unmarshal(body, &m)
+		if err != nil {
+			w.WriteHeader(http.StatusBadRequest)
+			w.Write([]byte(err.Error()))
+			return
+		}
+		if m.Name == "" {
+			w.WriteHeader(http.StatusBadRequest)
+			w.Write([]byte("You have to specify a name for the locally stored account."))
```

Comment:
this should be an adequate error

Reply:
that this should be an error from [types/errors.go](https://github.com/cosmos/cosmos-sdk/blob/develop/types/errors.go) and this line should only contain `w.Write([]byte(err.Error()))`. 

You can create another `sdk.Error` for to represent this logic

Reply:
I created client/key/errors.go instead. Because errors in types/errors.go are abci interfaced related.